server.go

```diff
@@ -2804,19 +2816,37 @@ func (s *server) fetchNodeAdvertisedAddr(pub *btcec.PublicKey) (net.Addr, error)
 	return node.Addresses[0], nil
 }
 
-// disableChannel disables a channel, resulting in it not being able to forward
-// payments. This is done by sending a new channel update across the network
-// with the disabled flag set.
-func (s *server) disableChannel(op wire.OutPoint) error {
+// announceChanStatus disables a channel if disabled=true, otherwise activates
+// it. This is done by sending a new channel update across the network with the
+// disabled flag set accordingly. The result of disabling the channel is it not
+// being able to forward payments.
+func (s *server) announceChanStatus(op wire.OutPoint, disabled bool) error {
+
+	// If we have already sent out an update reflecting the current status,
+	// skip this channel.
+	s.mu.RLock()
+	alreadyDisabled, ok := s.sentDisabled[op]
+	s.mu.RUnlock()
+	if ok && alreadyDisabled == disabled {
```

Comment:
with this locking procedure, we may send out duplicate announcements w/ same active/inactive status if `ok`. not sure how critical that is tho

If `!ok`, it would be possible to send out conflicting inactive/active statuses, and the network would see which ever finished last